### PR TITLE
Add the `local` case to the network string conversion function and narrow `Account` type to `Ed25519Account` in example

### PR DESCRIPTION
**Pull request by xbtmatt**
Currently you cannot use the wallet adapter in a local development environment with certain wallets because the middleware network string conversion fails, since it doesn't include the `local` case.

- Added the `local` case to the switch statement that converts a string to a Network.

### CODE REVIEW

File: apps/nextjs-example/components/transactionFlow/MultiAgent.tsx [hunk 1]
```diff
@@ -43,7 +43,7 @@ export default function MultiAgentTransaction({
       throw new Error("no network");
     }
 
-    const secondarySigner = Account.generate();
+    const secondarySigner = Ed25519Account.generate();
```
Comment (0xmaayan) [inline apps/nextjs-example/components/transactionFlow/MultiAgent.tsx]: nit:

dont really need this change as `Account.generate()` already creates a legacy Ed25519Account

File: apps/nextjs-example/components/transactionFlow/MultiAgent.tsx [hunk 2]
```diff
@@ -23,7 +23,7 @@ export default function MultiAgentTransaction({
     useWallet();
 
   const [secondarySignerAccount, setSecondarySignerAccount] =
-    useState<Account>();
+    useState<Ed25519Account>();
```
Comment (0xmaayan) [inline apps/nextjs-example/components/transactionFlow/MultiAgent.tsx]: this is definitely a needed change once the adapter is updated to use latest sdk version . check out https://github.com/aptos-labs/aptos-wallet-adapter/pull/296/files 😄

**Review by 0xmaayan — COMMENTED** [review]
let's remove the Account change in favor of the other PR that updates the SDK and holds this change already

File: apps/nextjs-example/components/transactionFlow/MultiAgent.tsx (hunk 1 above)
Comment (xbtmatt) [reply apps/nextjs-example/components/transactionFlow/MultiAgent.tsx]: Ah ok, I only did it to make it more explicit :p, wasn't sure if it'd type it correctly

**Review by 0xmaayan — APPROVED** [review]

**Pull request merged**